chain/store/store.go

```diff
@@ -1052,7 +1052,15 @@ func (cs *ChainStore) expandTipset(ctx context.Context, b *types.BlockHeader) (*
 		}
 
 		if cid, found := inclMiners[h.Miner]; found {
-			log.Warnf("Have multiple blocks from miner %s at height %d in our tipset cache %s-%s", h.Miner, h.Height, h.Cid(), cid)
+			log.Warnf("Have multiple blocks from miner %s at height %d in our tipset cache %s-%s, purging them all", h.Miner, h.Height, h.Cid(), cid)
+			newAll := make([]*types.BlockHeader, 0, len(all))
```

Comment:
We could ALSO drop the blocks from `cs.tipsets` entirely if we wanted to. I'm not sure this is the best thing to do, though, since that tracker is currently split between the "writer" method `AddToTipsetTracker` and the "reader" method `ExpandTipset`.

Reply:
I'd leave them so we can detect new ones.